Week3/homework/step2-6.js

```diff
@@ -2,15 +2,30 @@
 
 const arr2d = [[1, 2], [3, 4], [5, 6]];
 const arr3d = [[[1, 2], [3, 4]], [[5, 6], [7, 8]]];
+const flattenedArray = [];
 
-function flattenArray2d(arr) {
+function flattenArray2d() {
   // Replace this comment and the next line with your code
-  console.log(arr);
+  // console.log(arr);
+  for (let i = 0; i < arr2d.length; i++) {
+    for (let j = 0; j < arr2d[i].length; j++) {
+      console.log(arr2d[i][j]);
+    }
+  }
+  return flattenedArray;
 }
 
-function flattenArray3d(arr) {
+function flattenArray3d() {
   // Replace this comment and the next line with your code
-  console.log(arr);
+  // console.log(arr);
+  for (let x = 0; x < arr3d.length; x++) {
+    for (let y = 0; y < arr3d[x].length; y++) {
+      for (let z = 0; z < arr3d[x][y].length; z++) {
+        console.log(arr3d[x][y][z]);
+      }
+    }
+  }
+  return flattenedArray;
```

Comment:
Logic works fine but imagine if you have to do this for a 4-D or a 5-D array. What happens in that case? **Suggestion**: try reading about a programming concept called recursion and see if you can apply the same here. 🙂